packages/datadog-plugin-pg/test/index.spec.js

```diff
@@ -72,6 +72,39 @@ describe('Plugin', () => {
             })
           })
 
+          withVersions(plugin, 'pg-cursor', cursorVersion => {
```

Comment:
@rochdev I did an attempt adding this block mocking pg-cursor, I follow a bit the example of express plugin using loopback.
but I'm a bit lost, this doesn't seem to be executing in CI.

Reply:
That's odd. I wonder if it's caused by the native implementation not being supported by `pg-cursor` and maybe it causes the JS version to fail as well. I'd say to try to move this block after the parent describe (so just before the `with configuration` describe).

Reply:
no luck :( https://circleci.com/gh/DataDog/dd-trace-js/105068?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
It looks like `pg-cursor` expects `pg` to be available at the same level without explicitly depending on it. Since this is not the case in the test setup, the library fails to load and the test for it is skipped. I'm not sure how to support this, I'll have to think about it as it's an issue with how versions are installed for testing.

Reply:
ouch just noticed the same, thanks for the quick replies 👍 
seems that package should declare it at last as peerDependencies?
no rush, thanks

Reply:
It's designed to assume that you are installing both at the same level, which makes sense since it just adds an extension to `pg`. Ideally it would allow passing the `pg` module and not rely on the internals as it's doing, but unfortunately this is not the case.

I think I have an idea. It won't be pretty but hopefully it will work 😄 Trying it now.

Reply:
I opened #901 to allow defining peer dependencies at the instrumentation level. This will allow you to define `pg` as a peer in the config for `pg-cursor` like so:

```json
{
  "name": "pg-cursor",
  "versions": ["2.1.6"],
  "peers": ["pg"]
}
```

Since this will require a specific version of `pg`, the `withVersions` block for `pg-cursor` should also be moved out of the parent block since it will no longer need to run once for every version of `pg` but only for the version in `peers`.

Reply:
nice! thank you 👍

Reply:
I opened #903 as a replacement of #901 to do the above automatically without having to specify peers. Any module defined by an outer `withVersions` will always be available to an inner `withVersions`. This means you shouldn't have to change any test code and it should work after a rebase once the PR lands.

Were you able to use your branch successfully with a real app? I tried running the tests with the fix and it still doesn't seem to work, although it does load the `pg` module properly, so I'm not sure if I missed something or if there is an actual problem with the instrumentation.